index.js

```diff
@@ -15,7 +15,7 @@ module.exports = function(precompile) {
             throw file.errorWithNode(node, msg);
           }
 
-          this.remove();
+          this.dangerouslyRemove();
```

Comment:
Yes, that's a very good point. I hadn't thought about all the transitive dependencies, but a quick grep through a recent Ember project suggests that it `broccoli-babel-transpiler` is the package which has the dependency on babel-core, and it's only ^5.0.0.

I've pushed a revised commit. Not sure how you'd prefer to do the backwards compat check so I've used `typeof`, let me know if you'd like a different approach.